packages/frontend/model/validate.ts

```diff
@@ -1,26 +1,16 @@
 import Ajv from 'ajv';
+import { schema } from './schema/article.json.schema';
 
 export class ValidationError extends Error { }
 
 // enpoint can be used to reference matching schema
 export const validateRequestData = (data: any, endpoint: string) => {
-    const schema = {
-        properties: {
-            page: {
-                type: 'object',
-                properties: {
-                    content: {
-                        type: 'object',
-                        properties: {
-                            headline: { type: 'string' },
-                        },
-                    },
-                },
-            },
-        },
+    const options = {
+        verbose: true,
+        allErrors: true,
+        useDefaults: true, // modifies data in place
```

Comment:
This PR has now been update to make use of AJV's  built in `useDefaults` option that will fill in 'undefined' properties with defaults specified by the schema.  This should allow us to lose all of the `validator` functions currently in use in the extract functions a follow up PR

Reply:
Oh wow nice, that's really handy. I wonder if the frontend lib has something similar 🤔

Reply:
I've updated useDefaults: 'empty' which will also fallback to default options if a property is 'null | '''  This behaviour now matches best what our bespoke validators were doing. To note, if a property of non-string type in the JSON schema comes through as an empty string, it will useDefault and not through an error - is this desired behaviour?

This change is dependent on an update to the ajv type definition library however, in order to pass TeamCity.